pkg/pool-weighted/contracts/smart/IndexPool.sol

```diff
@@ -316,6 +316,26 @@ contract IndexPool is BaseWeightedPool, ReentrancyGuard {
         );
     }
 
+    /// @dev Hook is called when someone swaps through vault.
+    /// @param swapRequest Swap params.
+    /// @param currentBalanceTokenIn Vault balance of token that is swapped into the pool.
+    /// @param currentBalanceTokenOut Vault balance of token that is swapped out of the pool.
+    /// @return Amount of tokens that the user will receive in return for their token.
+    function onSwap(
+        SwapRequest memory swapRequest,
+        uint256 currentBalanceTokenIn,
+        uint256 currentBalanceTokenOut
+    ) public override returns (uint256) {
+        //cannot swap out uninitialized token
+        require(minBalances[swapRequest.tokenOut] == 0, "Uninitialized token");
```

Comment:
It should be `!= 0`, or am I wrong?

Reply:
No, we are 
1. checking which token is to be swapped out of the pool
2. if it is an unitialized/new token it will have a minimumBalance set
3. in this case we want to revert
Phrase differently we require the minimumBalance of the token that is swapped out to be zero, which means that it is not a new token.

Reply:
Hmm, old tokens have minimumBalance as well I think, no?

Reply:
We get them passed as params in `reindexTokens` (because the controller is dumb) but we don't do anything with them, since we don't need them. We only store the minBalance for new tokens, everything else would also be a huge waste of gas. We should just make sure that we also delete it when we remove a token from the pool.

Reply:
https://github.com/PrimeDAO/balancer-v2-monorepo/blob/feature/53/uninitialized-token-behavior/pkg/pool-weighted/contracts/smart/IndexPool.sol#L290